cpp/src/neighbors/sample_filter.cuh

```diff
@@ -140,4 +146,48 @@ void bitmap_filter<bitmap_t, index_t>::to_csr(raft::resources const& handle, csr
   raft::sparse::convert::bitmap_to_csr(handle, bitmap_view_, csr);
 }
 
+struct none_filter_args_t {};
+using bitset_filter_args_t =
+  std::tuple<const int64_t* const*, cuvs::core::bitset_view<uint32_t, int64_t>>;
+
+struct ivf_filter_dev {
+  filtering::FilterType tag_;
+
+  union ivf_filter_dev_args_variant {
+    none_filter_args_t none_filter_args;
+    bitset_filter_args_t bitset_filter_args;
+
+    _RAFT_HOST_DEVICE ivf_filter_dev_args_variant() {}
+    _RAFT_HOST_DEVICE ~ivf_filter_dev_args_variant() {}
```

Comment:
Can we get away with no explicitly defined constructor/destructor? I have a feeling this looks more complicated than should :)

Reply:
The compiler complains if I try to do this. 
```
error: the default constructor of "cuvs::neighbors::filtering::ivf_filter_dev::ivf_filter_dev_args_variant" cannot be referenced -- it is a deleted function
```

Looks like this is because the `bitset_filter_args_t` is **not** trivially constructible.

Reply:
What line is this error occurring on?

Reply:
line 159 and 164 of `sample_filter.cuh`.  That would be the `ivf_filter_dev` constructors.

```
sample_filter.cuh(159): error: the default constructor of "cuvs::neighbors::filtering::ivf_filter_dev::ivf_filter_dev_args_variant" cannot be referenced -- it is a deleted function
    {
    ^

/home/nfs/jinsolp/rapids/cuvs/cpp/src/neighbors/ivf_flat/../sample_filter.cuh(164): error: the default constructor of "cuvs::neighbors::filtering::ivf_filter_dev::ivf_filter_dev_args_variant" cannot be referenced -- it is a deleted function
    {
    ^
```